File: discord-bot/app/src/main/kotlin/es/magonxesp/pekorabot/modules/shared/infraestructure/strapi/file/File.kt
package es.magonxesp.pekorabot.modules.shared.infraestructure.strapi.file


import kotlinx.serialization.Serializable

@Serializable
data class File(
    val id: Int,
    val attributes: FileAttributes
) {
    @Serializable
    data class FileAttributes(
        val url: String
    )
}

File: app/frontend/modules/shared/infraestructure/strapi/file.ts
export interface FileAttributes {
  url: string
}

export interface File {
  id?: number
  attributes: FileAttributes
}

export interface FileData {
  data: File
}

export function fileUrlToFileData(url?: string): FileData | null {
  if (!url) {
    return null
  }

  return {
    data: {
      attributes: {
        url: url
      }
    }
  }
}
